### PR TITLE
Fix: do not change the workflow after the env changed

### PR DESCRIPTION
**Pull request by barnettZQG**
Signed-off-by: barnettZQG <barnett.zqg@gmail.com>

### Description of your changes

Enable users to edit the workflow to determine targets for the application to be published to the environment.

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/apiserver/domain/service/env.go [hunk 1]
```diff
@@ -222,9 +205,7 @@ func (p *envServiceImpl) UpdateEnv(ctx context.Context, name string, req apisv1.
 		return nil, bcode.ErrEnvTargetConflict
 	}
 
-	var targetChanged bool
 	if len(req.Targets) > 0 && !checkEqual(env.Targets, req.Targets) {
```
Comment (StevenLeiZhang) [inline pkg/apiserver/domain/service/env.go:208]: Shall we check if any APP exists(deployed) in this ENV? If this ENV already has APP, reject the update request.
That means user need delete APP before updating ENV's target.

Comment (barnettZQG) [reply pkg/apiserver/domain/service/env.go:208]: I think we do not need to check the app when adding a new target to the environment. For deleting the targets, maybe we should check. What do you think?

Comment (StevenLeiZhang) [reply pkg/apiserver/domain/service/env.go:208]: > I think we do not need to check the app when adding a new target to the environment. For deleting the targets, maybe we should check. What do you think?

Yes, I think
1. if user only update 'description' or 'alias', no need pre-check.
2. if user want to remove some targets from ENV, need pre-check.
3. if user want to add more targets, no need pre-check.

Comment (barnettZQG) [reply pkg/apiserver/domain/service/env.go]: @StevenLeiZhang Could review again.

**Review by FogDong — APPROVED** [review]

**Pull request merged**